MobileBuy/buy/src/main/java/com/shopify/buy3/HttpCallbackWithRetry.java

```diff
@@ -125,8 +125,8 @@ private void notifyError(final GraphError error) {
     Runnable action = () -> {
       if (error instanceof GraphNetworkError) {
```

Comment:
We give option to user to react differently to different errors, see `com.shopify.buy3.GraphCall.Callback`.

By default if user doesn't care about the specific type he will be notified with general error in `com.shopify.buy3.GraphCall.Callback#onFailure`

If you look closely to `com.shopify.buy3.GraphCall.Callback#onHttpError`:
```
try {
        onFailure(error);
      } finally {
        error.dispose();
      }
```
For `GraphHttpError` we give user a chance to get access to raw okHttp response that must be closed to avoid the leaking resource.